clippy_lints/src/derive.rs

```diff
@@ -163,12 +163,11 @@ fn check_hash_peq<'a, 'tcx>(
                         cx, DERIVE_HASH_XOR_EQ, span,
                         mess,
                         |diag| {
-                        if let Some(hir_id) = cx.tcx.hir().as_local_hir_id(impl_id) {
-                            diag.span_note(
-                                cx.tcx.hir().span(hir_id),
-                                "`PartialEq` implemented here"
-                            );
-                        }
+                        let hir_id = cx.tcx.hir().as_local_hir_id(impl_id.expect_local());
```

Comment:
I would prefer `if let Some(hir_id) = impl_id.as_local()`, since `expect_local` could panic, where the code previously couldn't.

Also formatting: +1 indent.